tests/test.nix

```diff
@@ -25,14 +48,62 @@ in makeTest' {
       # speed-up eval
       documentation.enable = false;
 
-      virtualisation.emptyDiskImages = [ 512 512 ];
+      virtualisation.diskSize = 512;
+      virtualisation.emptyDiskImages = [ 32 32 ];
     };
+in makeTest' {
+  name = "disko";
+
+  nodes.manual1 = baseConfig;
+
+  nodes.automatic1 = {pkgs, modulesPath, ...}: {
+    imports = [
+      baseConfig
+      (modulesPath + "/installer/cd-dvd/installation-cd-minimal.nix")
+    ];
+    systemd.services.install-to-hd = {
+      enable = true;
+      wantedBy = ["multi-user.target"];
+      after = ["getty@tty1.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        ExecStart = [
+          (wrapScript {inherit pkgs; script = tsp-create;})
+          (wrapScript {inherit pkgs; script = tsp-mount;})
+        ];
+        StandardInput = "null";
+        StandardOutput = "journal+console";
+        StandardError = "inherit";
+      };
+    };
+  };
 
   testScript = ''
-    machine.succeed("echo 'secret' > /tmp/secret.key");
-    machine.succeed("${tsp-create}");
-    machine.succeed("${tsp-mount}");
-    machine.succeed("${tsp-mount}"); # verify that the command is idempotent
-    machine.succeed("test -b /dev/md/raid1");
+    EXPECTED_BLKS = """
+    vda 253:0 0 512M 0 disk /
+    vdb 253:16 0 32M 0 disk
+    vdb1 253:17 0 31M 0 part
+    md127 9:127 0 29.9M 0 raid1
+    md127p1 259:0 0 28.9M 0 part
+    vdc 253:32 0 32M 0 disk
+    vdc1 253:33 0 31M 0 part
+    md127 9:127 0 29.9M 0 raid1
+    md127p1 259:0 0 28.9M 0 part
+    """
+
+    start_all()
+
+    manual1.succeed("${tsp-create}");
+    manual1.succeed("${tsp-mount}");
+    manual1.succeed("${tsp-mount}"); # verify that the command is idempotent
+
+    for node in [manual1, automatic1]:
+      node.wait_for_file("/raid")
```

Comment:
If you run the tests now, you'll see `automatic1` stalls in this line. That should mean that there's something wrong when setting up a raid with a systemd unit.

Some help here please?

Reply:
I guess it gets mounted to /mnt/raid (since it's really meant for a nixos install, maybe that should be more clearer in some doc)

Reply:
I don't understand... According to this line, it seems like it should mount on `/raid`, right?

https://github.com/nix-community/disko/blob/6b0b20da18cdffd09f04faee7128c557bcb9f054/example/raid.nix#L52

Besides, on machine `manual1` this check succeeds.

Reply:
huh, well we prepend `/mnt` on most paths. unsure why it should succeed on manual1 https://github.com/nix-community/disko/blob/6b0b20da18cdffd09f04faee7128c557bcb9f054/default.nix#L133